File: site/src/lib/modules/question/scripts/models/question.ts
import { Answer, RawAnswer } from "./answer";
import { QuestionText } from "../value_objects/question_text";
import { RawUser, User } from "../../../user/scripts/models/user";

// type Question struct {
// 	QuestionId   id.ID               `json:"question_id"`
// 	CreatorId    id.ID               `json:"creator_id"`
// 	QuestionText models.QuestionText `json:"question_text"`
// 	Answer       Answer              `json:"answer"`
// 	CreatedAt    time.Time           `json:"created_at"`
// }

export class Question {
  questionId: string;
  creator: User;
  questionText: QuestionText;
  answer: Answer;
  createdAt: Date;
}

export class RawQuestion {
  question_id: string;
  creator: RawUser;
  question: string;
  answer: RawAnswer;
  created_at: string;

  constructor(question: string, answer: RawAnswer) {
    this.question = question;
    this.answer = answer;
  }
}

export function NewQuestion() { }

File: api/src/app/questions/models/question.go
package models

import (
	userModels "lib/app/users/models"
	"lib/common/id"
	"time"
)

type Question struct {
	QuestionId   id.ID           `gorm:"primaryKey"`
	CreatorId    id.ID           `gorm:"column:creator_id"`
	Creator      userModels.User `gorm:"foreignKey:user_id;references:creator_id"`
	QuestionText QuestionText    `gorm:"column:question_text"`
	Answer       Answer          `gorm:"column:answer"`
	CreatedAt    time.Time       `gorm:"column:created_at"`
}

func NewQuestion(id id.ID, creatorId id.ID, questionText QuestionText, answer Answer, createdAt time.Time) Question {
	return Question{
		QuestionId:   id,
		CreatorId:    creatorId,
		QuestionText: questionText,
		Answer:       answer,
		CreatedAt:    createdAt,
	}
}
